packages/cactus-plugin-ledger-connector-corda/src/main/typescript/plugin-ledger-connector-corda.ts

```diff
@@ -65,6 +67,26 @@ export class PluginLedgerConnectorCorda
   > {
     return ConsensusAlgorithmFamily.AUTHORITY;
   }
+  public async hasTransactionFinality(): Promise<boolean> {
+    const currentConsensusAlgorithmFamily = await this.getConsensusAlgorithmFamily();
+
+    const unrecognizedConsensusAlgorithmFamily =
```

Comment:
@jscode017 The part below could be pulled out into a utility function that could be placed in the `cactus-core` package (not `cactus-core-api`).

```typescript
  const unrecognizedConsensusAlgorithmFamily =
      !(Object.values(
        ConsensusAlgorithmFamiliesWithTxFinality,
      ) as string[]).includes(currentConsensusAlgorithmFamily.toString()) &&
      !(Object.values(
        ConsensusAlgorithmFamiliesWithOutTxFinality,
      ) as string[]).includes(currentConsensusAlgorithmFamily.toString());

    if (unrecognizedConsensusAlgorithmFamily) {
      throw new Error(
        `Unrecognized consensus algorithm family: ${currentConsensusAlgorithmFamily}`,
      );
    }
    return (Object.values(
      ConsensusAlgorithmFamiliesWithTxFinality,
    ) as string[]).includes(currentConsensusAlgorithmFamily.toString());
```

Then the `hasTransactionFinality()` methods could all just reuse that method instead of duplicating the same code in every connector.

Reply:
Yeah, I agree that pulled it out would be a much better way. 
But which class should I put the function to? Both ConsortiumRepository and PluginRegistry in cactus-core do not seems to be a good place?

Reply:
BTW, my wrap up function may look like this:
```
public consensusHasTransactionFinality(consensusAlgorithmFamily:ConsensusAlgorithmFamily) boolean{
    const isInConsensusAlgorithmFamiliesWithTxFinality=(Object.values(     ConsensusAlgorithmFamiliesWithTxFinality ) as string[]).includes(consensusAlgorithmFamily.toString());
    
    const isInConsensusAlgorithmFamiliesWithOutTxFinality=(Object.values(  ConsensusAlgorithmFamiliesWithOutTxFinality) as string[]).includes(consensusAlgorithmFamily.toString());
    
    const unrecognizedConsensusAlgorithmFamily =! isInConsensusAlgorithmFamiliesWithTxFinality && !isInConsensusAlgorithmFamiliesWithOutTxFinality;

    if (unrecognizedConsensusAlgorithmFamily) {
      throw new Error(
        `Unrecognized consensus algorithm family: ${consensusAlgorithmFamily}`,
      );
    }
    return isConsensusAlgorithmFamiliesWithTxFinality;
    }
```

Reply:
Sorry I get a stupid question here.
I add 
`export { consensusHasTransactionFinality } from "./consensus-has-transaction-finality";`
or
`export * from "./consensus-has-transaction-finality";`
But keep getting
`src/main/typescript/public-api.ts(7,49): error TS2307: Cannot find module './consensus-has-transaction-finality' or its corresponding type declarations.`

Reply:
Hmm, that's strange indeed. Can you push to the remote so I can check it out as well?

Reply:
Sure , I've pushed it to the remote.

Reply:
Ran the build on it, to me it was saying that you are missing `ConsensusAlgorithmFamily` from your import statement (which then makes that file not compile and that leads to the compilation of public-api.ts bugging out.

This is how I got it to build:

```typescript
import {
  ConsensusAlgorithmFamiliesWithTxFinality,
  ConsensusAlgorithmFamiliesWithOutTxFinality,
  ConsensusAlgorithmFamily,
} from "@hyperledger/cactus-core-api";

export function consensusHasTransactionFinality(
  consensusAlgorithmFamily: ConsensusAlgorithmFamily,
): boolean {
  const isInConsensusAlgorithmFamiliesWithTxFinality = (Object.values(
    ConsensusAlgorithmFamiliesWithTxFinality,
  ) as string[]).includes(consensusAlgorithmFamily.toString());

  const isInConsensusAlgorithmFamiliesWithOutTxFinality = (Object.values(
    ConsensusAlgorithmFamiliesWithOutTxFinality,
  ) as string[]).includes(consensusAlgorithmFamily.toString());

  const unrecognizedConsensusAlgorithmFamily =
    !isInConsensusAlgorithmFamiliesWithTxFinality &&
    !isInConsensusAlgorithmFamiliesWithOutTxFinality;

  if (unrecognizedConsensusAlgorithmFamily) {
    throw new Error(
      `Unrecognized consensus algorithm family: ${consensusAlgorithmFamily}`,
    );
  }
  return isInConsensusAlgorithmFamiliesWithTxFinality;
}
//export defualt consensusHasTransactionFinality;

```

Reply:
Almost finish.
But there is a small issue remains.
Would want to know how to add dependencies for this project.
I have uploaded a new commit, but somehow cound not refactor "cactus-plugin-ledger-connector-corda".
I would want like to add "@hyperledger/cactus-core-api": "0.4.1" in cactus-plugin-ledger-connector-corda/package.json. 
But instead break dependencies and got bunch of things like below:
```
src/main/typescript/plugin-factory-ledger-connector.ts(4,8): error TS2307: Cannot find module '@hyperledger/cactus-core-api' or its corresponding type declarations.
src/main/typescript/plugin-ledger-connector-corda.ts(15,8): error TS2307: Cannot find module '@hyperledger/cactus-core-api' or its corresponding type declarations.
src/main/typescript/plugin-ledger-connector-corda.ts(16,10): error TS2305: Module '"../../../node_modules/@hyperledger/cactus-core/dist/types/main/typescript"' has no exported member 'consensusHasTransactionFinality'.
src/main/typescript/plugin-ledger-connector-corda.ts(22,8): error TS2307: Cannot find module '@hyperledger/cactus-common' or its corresponding type declarations.
src/main/typescript/plugin-ledger-connector-corda.ts(55,27): error TS2339: Property 'instanceId' does not exist on type 'IPluginLedgerConnectorCordaOptions'.
src/main/typescript/plugin-ledger-connector-corda.ts(60,36): error TS2339: Property 'instanceId' does not exist on type 'IPluginLedgerConnectorCordaOptions'.
src/main/typescript/public-api.ts(15,39): error TS2307: Cannot find module '@hyperledger/cactus-core-api' or its corresponding type declarations.
src/main/typescript/public-api.ts(21,43): error TS2554: Expected 0 arguments, but got 1.
src/main/typescript/web-services/deploy-contract-jars-endpoint.ts(11,8): error TS2307: Cannot find module '@hyperledger/cactus-core-api' or its corresponding type declarations.
src/main/typescript/web-services/deploy-contract-jars-endpoint.ts(18,8): error TS2307: Cannot find module '@hyperledger/cactus-common' or its corresponding type declarations.
src/test/typescript/integration/deploy-cordapp-jars-to-nodes.test.ts(4,33): error TS2307: Cannot find module '@hyperledger/cactus-test-tooling' or its corresponding type declarations.
src/test/typescript/integration/deploy-cordapp-jars-to-nodes.test.ts(5,30): error TS2307: Cannot find module '@hyperledger/cactus-common' or its corresponding type declarations.
src/test/typescript/integration/deploy-cordapp-jars-to-nodes.test.ts(9,8): error TS2307: Cannot find module '@hyperledger/cactus-test-tooling' or its corresponding type declarations.
src/test/typescript/integration/jvm-kotlin-spring-server.test.ts(7,8): error TS2307: Cannot find module '@hyperledger/cactus-test-tooling' or its corresponding type declarations.
src/test/typescript/integration/jvm-kotlin-spring-server.test.ts(8,30): error TS2307: Cannot find module '@hyperledger/cactus-common' or its corresponding type declarations.
src/test/typescript/integration/jvm-kotlin-spring-server.test.ts(12,8): error TS2307: Cannot find module '@hyperledger/cactus-test-tooling' or its corresponding type declarations.
```